src/Engine/Objects/Player.cpp

```diff
@@ -3082,18 +3103,30 @@ PLAYER_SKILL_LEVEL Player::GetActualSkillLevel(PLAYER_SKILL_TYPE uSkillType) {
 }
 
 PLAYER_SKILL_MASTERY Player::GetActualSkillMastery(PLAYER_SKILL_TYPE uSkillType) {
-    return GetSkillMastery(uSkillType);
+    PLAYER_SKILL_MASTERY value = GetSkillMastery(uSkillType);
+
+    if (shouldSkillBeShared(uSkillType)) {
+        for (Player &player : pParty->pPlayers) {
+            if (&player != this && player.CanAct()) {
+                PLAYER_SKILL_MASTERY otherPlayerResult = player.GetSkillMastery(uSkillType);
+                value = std::max(value, otherPlayerResult);
+            }
+        }
+    }
```

Comment:
Also we may want to add as a TODO - returning which players mastery get used. So that reactions come from the correct player.